www/common/inner/common-modal.js

```diff
@@ -49,7 +49,13 @@ define([
                         // Access modal and the pad is not stored: get the hashes from outer
                         var hashes = priv.hashes || {};
                         // For calendars, individual href is passed via opts
-                        data.href = ((priv.app === 'calendar') && opts.href) || Hash.hashToHref(hashes.editHash || hashes.fileHash, priv.app);
+                        if (priv.app === 'calendar') {
+                            data.href = (priv.app === 'calendar') && opts.href;
+                        } else if (hashes.editHash || hashes.fileHash) {
+                            data.href = Hash.hashToHref(hashes.editHash || hashes.fileHash);
```

Comment:
You forgot `priv.app` as the second argument so we end up with the "undefined" app:
`http://localhost:3000/undefined/#/2/code/edit/DRlul1bzt7FpOdlzik8ZLA2F/`